src/main/java/com/github/scr/j8iterables/core/CloseableSpliterator.java

```diff
@@ -0,0 +1,72 @@
+package com.github.scr.j8iterables.core;
+
+import javax.annotation.Nullable;
+import java.io.Closeable;
+import java.util.Comparator;
+import java.util.Spliterator;
+import java.util.function.Consumer;
+
+/**
+ * A {@link Spliterator} that closes a {@link Closeable} resource when it is completed.
+ *
+ * @author scr on 2/24/17.
+ * @apiNote Upon close, any {@link java.io.IOException} is converted to {@link java.io.UncheckedIOException}.
+ */
+public class CloseableSpliterator<T, R> implements Spliterator<T> {
```

Comment:
I wish there is a type specification for R = `AutoClosable`, then `resourceCloser` simply becomes `resource::close`.